src/utils/helpers.ts

```diff
@@ -1,14 +1,18 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
+
 import * as reselect from 'reselect';
 import shallowEqual from 'shallow-equals';
+
 import {Dictionary} from 'types/utilities';
-export function memoizeResult<F extends Function>(func: F): any {
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export function memoizeResult<F extends Function>(func: F): F {
```

Comment:
This is the only case where I couldn't figure out how to get rid of `Function`, so we have to do some ugly casting through `unknown` below as well.